packages/tenant-process/src/services/tenantService.ts

```diff
@@ -261,6 +288,87 @@ export function tenantServiceBuilder(
       }
     },
 
+    async addCertifiedAttribute(
+      {
+        tenantId,
+        tenantAttributeSeed,
+        organizationId,
+        correlationId,
+      }: {
+        tenantId: TenantId;
+        tenantAttributeSeed: tenantApi.CertifiedTenantAttributeSeed;
+        organizationId: TenantId;
+        correlationId: string;
+      },
+      logger: Logger
+    ): Promise<Tenant> {
+      logger.info(
+        `Add certified attribute ${tenantAttributeSeed.id} to tenant ${tenantId}`
+      );
+
+      const requesterTenant = await retrieveTenant(
+        organizationId,
+        readModelService
+      );
+
+      const certifierId = getTenantCertifierId(requesterTenant.data);
+
+      const attribute = await retrieveAttribute(
+        unsafeBrandId(tenantAttributeSeed.id),
+        readModelService
+      );
+
+      if (attribute.kind !== attributeKind.certified) {
+        throw attributeNotFound(attribute.id);
+      }
+
+      if (!attribute.origin || attribute.origin !== certifierId) {
+        throw attributeDoesNotBelongToCertifier(
+          attribute.id,
+          organizationId,
+          tenantId
+        );
+      }
+
+      const targetTenant = await retrieveTenant(tenantId, readModelService);
+
+      const tenantWithNewAttribute = await assignCertifiedAttribute({
+        targetTenant: targetTenant.data,
+        attribute,
+      });
+
+      const [updatedTenant, event] = await revaluateTenantKind({
+        tenant: tenantWithNewAttribute,
+        version: targetTenant.metadata.version,
+        readModelService,
+        correlationId,
+      });
+
+      const tenantCertifiedAttributeAssignedEvent =
+        toCreateEventTenantCertifiedAttributeAssigned(
+          targetTenant.metadata.version,
+          updatedTenant,
+          attribute.id,
+          correlationId
+        );
+
+      if (event) {
+        const tenantKindUpdatedEvent: CreateEvent<TenantEvent> = {
+          ...event,
+          version: targetTenant.metadata.version + 1,
+        };
```

Comment:
Please don't modify the event after it has been packed by the function `toCreateEvent{...}`

Reply:
You could already pass the increased version.